### PR TITLE
backport - fixed checking if proxy and username pass are empty

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Backport of https://github.com/xbmc/xbmc/pull/17338

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
Win32 failure unrelated.

File: xbmc/filesystem/CurlFile.cpp [hunk 1]
```diff
@@ -603,7 +603,7 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
 
     const std::string userpass =
       m_proxyuser + std::string(":") + m_proxypassword;
-    if (!userpass.empty())
+    if (userpass != ":")
```
Comment (howie-f) [inline xbmc/filesystem/CurlFile.cpp:606]: `if (StringUtils::StartsWith(userpass, ":") || StringUtils::EndsWith(userpass, ":")) ....`
only one or both empty?

Comment (phunkyfish) [reply xbmc/filesystem/CurlFile.cpp:606]: The original intent was to signify both were empty. I don’t know if one or the other is valid in curl. Maybe it is.

Comment (phunkyfish) [reply xbmc/filesystem/CurlFile.cpp:606]: Ok, I've done a bit of reading on this. Turns out that user/pass needs to have both and they also need to be separately url encoded and then joined with a `:`.

So I think we merge this PR as is (as this is already in Matrix) and then do new Matrix PR fixing it up correctly. Checking for both and the URL encoding. Would you like to do that PR @howie-f ?

Comment (howie-f) [reply xbmc/filesystem/CurlFile.cpp:606]: yes, did some reading too, but haven't found anything really usable. we can do as you suggest (yet i don't unterstand what is meant by "separately url encoded", but i'll figure out). will then open the matrix-fix this evening when i'm back home

Comment (phunkyfish) [reply xbmc/filesystem/CurlFile.cpp:606]: Do you want to approve this one so and we'll put it in?

Separately URL encoded means something like this in pseudocode: `URLEncode(username) + ":" + URLEncode(password)`. Maybe `%3A` substitutions for any `:`'s would also work.

**Review by howie-f — APPROVED** [review]
looks good

**Pull request merged**